test/Path/Gen.hs

```diff
@@ -45,7 +45,6 @@ instance Validity (Path Rel Dir) where
     && FilePath.hasTrailingPathSeparator fp
     && FilePath.isValid fp
     && not (null fp)
-    && fp /= "."
     && fp /= ".."
     && not (".." `isInfixOf` fp)
     && (parseRelDir fp == Just p)
```

Comment:
Ah, it seems I removed the wrong line! It still cannot be "." but it can be null. So we need to remove the `not (null fp)`  condition and keep the `fp /= "."` condition.